src/ModuleList/Quiz/Quiz.tsx

```diff
@@ -1,22 +1,78 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useParams } from 'react-router-dom';
+import { SimpleGrid, Box, Flex } from '@chakra-ui/core';
+import { SubmitQuiz, CheckIcon, IncorrectIcon } from './styled';
+import { useQuizMeta } from 'src/common/api/curriculum/quiz';
+import { Question } from 'src/common/api/curriculum/quiz/requests';
 
 interface Params {
-  moduleId: string;
   quizId: string;
 }
 
+interface StringMap {
+  [key: string]: boolean;
+}
+
 const Quiz = () => {
-  const { moduleId, quizId } = useParams<Params>();
+  const { quizId } = useParams<Params>();
+
+  const { data } = useQuizMeta(Number(quizId));
+
+  const [questionIndexesAnsweredCorrectly, setQuestionIndexesAnsweredCorrectly] = useState<StringMap>({});
+  const [quizIsSubmitted, setQuizIsSubmitted] = useState<boolean>(false);
+
+  const handleAnsweredQuestion = (q: Question, qIndex: number, optionIndex: number) => {
+    return (e: React.ChangeEvent<any>) => {
+      questionIndexesAnsweredCorrectly[qIndex] = q.answer === optionIndex;
+      setQuestionIndexesAnsweredCorrectly(questionIndexesAnsweredCorrectly);
+    };
+  };
+
+  const checkQuiz = () => {
+    setQuizIsSubmitted(true);
+  };
+
+  const getLabelStyle = (q: Question, qIndex: number, optionIndex: number) => {
+    let color = '#FFFFFF';
+    if (quizIsSubmitted &&
+      !!questionIndexesAnsweredCorrectly[qIndex] === false &&
+      q.answer === optionIndex) {
+      color = '#C5A3B4';
+    }
+    return { color };
+  };
 
   return (
     <div>
-      <h1>
-        Module: {moduleId}
-      </h1>
-      <h2>
-        Quiz: {quizId}
-      </h2>
+      <SimpleGrid columns={1} spacing={10}>
+        {Array.isArray(data) && data.map((q: Question, qIndex: number) => (
+          <Box p={5} color="white" key={qIndex}>
```

Comment:
We probably shouldn't use indexes as keys. See my other comment about this. Maybe try `q.question`?

```suggestion
          <Box p={5} color="white" key={q.question}>
```